dyn_sim/sys/planar/segway.py

```diff
@@ -1 +1,218 @@
+    Input vector is in R.
+        u - motor voltage
+    """
+
+    def __init__(
+        self,
+        m0: float = 52.71,
+        L: float = 0.169,
+        J0: float = 5.108,
+        Km: float = 2 * 1.262,
+        R: float = 0.195,
+        bt: float = 2 * 1.225,
+        l: float = 0.75,
+    ) -> None:
+        """Initialize a segway.
+
+        Parameters
+        ----------
+        m0 : float
+            Lumped mass of the segway
+        L : float
+            Length between the center of rotation and center of gravity
+        J0 : float
+            Lumped inertia of the segway
+        Km : float
+            Motor Torque constant
+        R : float
+            Wheel radius
+        bt : float
+            Motor damping constant
+        l : float
+            Length between center of rotation and tip of segway arm
+        """
+        super(Segway, self).__init__(4, 1, False)
+        self._m0 = m0
+        self._L = L
+        self._J0 = J0
+        self._Km = Km
+        self._R = R
+        self._bt = bt
+        self._l = l
+
+    def fdyn(self, t: float, s: np.ndarray) -> np.ndarray:
+        """Segway autonomous dynamics.
+
+        Parameters
+        ----------
+        t : float,
+            Time. Unused, included for API compliance.
+        s : np.ndarray, shape=(4,)
+            State of segway.
+
+        Returns
+        -------
+        _fdyn : np.ndarray, shape=(4,)
+            Time derivatives of states from autonomous dynamics.
+        """
+        assert s.shape == (4,)
+
+        # states
+        # p = s[0] # unused
+        phi = s[1]
+        dp = s[2]
+        dphi = s[3]
+
+        sinphi = np.sin(phi)
+        cosphi = np.cos(phi)
+
+        # inertia matrix
+        D = np.array(
+            [
+                [self._m0, self._m * self._L * cosphi],
+                [self._m * self._L * cosphi, self._J0],
+            ]
+        )
+        Dinv = np.linalg.inv(D)
+
+        # coriolis & gravity matrix
+        H = np.array(
+            [
+                -self._m * self._L * sinphi * dphi**2
+                + self._bt * (dp - self._R * dphi) / self._R,
+                -self._m * self._g * self._L * sinphi
+                + self._bt * (dp - self._R * dphi),
+            ]
+        )
+
+        DinvH = Dinv @ H
+        _fdyn = np.array([dp, dphi, -DinvH[0], -DinvH[1]])
+        return _fdyn
+
+    def gdyn(self, t: float, s: np.ndarray) -> np.ndarray:
+        """Segway control dynamics.
+
+        Parameters
+        ----------
+        t : float,
+            Time. Unused, included for API compliance.
+        s : np.ndarray, shape=(4,)
+            State of segway.
+
+        Returns
+        -------
+        _gdyn : np.ndarray, shape=(4,)
+            Matrix representing affine control dynamics.
+        """
+        assert s.shape == (4,)
+
+        # states
+        # p = s[0] # unused
+        phi = s[1]
+        # dp = s[2] # unused
+        # dphi = s[3] # unused
```

Comment:
probably don't need to specify this many unused, clutters the code. state already explained in the class docstring.